entrypoint.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -e
+
+# Remove a potentially pre-existing server.pid for Rails.
+rm -f /myapp/tmp/pids/server.pid
+
+# Ensure all gems are installed.
+bundle check || bundle install
+
+# Precompile assets (optional for development, but useful for production)
+bundle exec rails assets:precompile
```

Comment:
This is better done within the dockerfile.

Reply:
And the thing that you _aren't_ doing in the entrypoint, that ends up being important for a production deploy, is database migrations.

it's probably worth checking out the default dockerfile that rails 7.1+ generates now.  There are some tricks at play.

Reply:
Ah yes same for here - I am not sure we are using this for production deployments right now. I was thinking this is for dev/test to get a developer up and running only. I need to read up more on deploying with docker - I am not sure why we would need to do this in the first place!

Reply:
If you're deploying the image -- the precompile _should_ be done during the docker build, not during startup.

It's less of an issue if you're just using importmaps, I _think_  (I'm not overly familiar with how they work), but generally the pattern is to use a build container to precompile your assets, and then copy the assets over to the production image.  Rationale is that the production image then _only_ needs to ship with ruby/database libraries, and doens't need to cart along a node installation and 400+ `node_modules` packages ...

which isn't _just_ deployment image size, it's also attack area.

Reply:
Ok I 'think' that we are currently doing this by calling asset precompile command in the dockerfile, no?

Reply:
yep